Python/Product/TestAdapter.Executor/Config/PythonProjectSettings.cs

```diff
@@ -39,6 +40,7 @@ public class PythonProjectSettings : IEquatable<PythonProjectSettings> {
             Enum.TryParse<TestFrameworkType>(testFramework, ignoreCase: true, out TestFramwork);
             UnitTestPattern = string.IsNullOrEmpty(unitTestPattern) ? PythonConstants.DefaultUnitTestPattern : unitTestPattern;
             UnitTestRootDir = string.IsNullOrEmpty(unitTestRootDir) ? PythonConstants.DefaultUnitTestRootDirectory : unitTestRootDir;
+            DiscoveryWaitTimeInSeconds = Int32.TryParse(discoveryWaitTimeInSeconds, out int parsedWaitTime) ? parsedWaitTime : PythonConstants.DiscoveryTimeoutInSeconds;
         }
 
```

Comment:
making Discovery wait time configurable through the test settings so we can change it in tests